### PR TITLE
Add EIP-5791: Physical Backed Tokens

### PR DESCRIPTION
**Pull request by 2pmflow**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

**Review by Gozzeella — DISMISSED** [review]
Innovative approach to bridge web3 and physical world

**Comment by ivanmmurciaua** [comment]
That's awesome draft and very interesting real value for decentralized world

**Comment by edddyguo** [comment]
Awesome project！

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
A lot of the content of the spec is actually motivation. Please move the motivation into the motivation section.

**Comment by 2pmflow** [comment]
@Pandapip1 I moved a chunk to Motivation here https://github.com/ethereum/EIPs/pull/5791/commits/a14103cac4cf6232fce2647c72942f63e04ce460, but was having trouble finding other parts of the spec that are a better fit in Motivation than their current respective sections, based on the prompt for Motivation in the template (`The motivation section should describe the "why" of this EIP.`). To me, the "why" is to solve for the two enumerated problems, and the rest is "how" the spec solves those problems. Could you clarify what chunks you think are a better fit for the "why"? Happy to make changes as appropriate.

**Comment by YAYLABS-Snoo** [comment]
I think it will help develop various interesting contents that can be done offline as well as trading.

File: EIPS/eip-5791.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-10-17
+requires: 191, 721
+---
+
+## Abstract
+
+This standard is an extension of [EIP-721](./eip-721.md). It proposes a minimal interface for an [EIP-721](./eip-721.md) NFT to be "physically backed" and owned by whoever owns the NFT's physical counterpart.
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:16]:
```suggestion
This standard is an extension of [EIP-721](./eip-721.md). It proposes a minimal interface for a [EIP-721](./eip-721.md) NFT to be "physically backed" and owned by whoever owns the NFT's physical counterpart.
```

File: EIPS/eip-5791.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+## Motivation
+
+NFT collectors enjoy collecting digital assets and sharing them with others online. However, there is currently no such standard for showcasing physical assets as NFTs with verified authenticity and ownership. Existing solutions are fragmented and tend to be susceptible to at least one of the following:
+
+- The NFT cannot proxy as ownership of the physical item. In most current implementations, the NFT and physical item are functionally two decoupled distinct assets after the NFT mint, in which the NFT can be freely traded independently from the physical item.
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:22]:
```suggestion
- The ownership of the physical item and the ownership of the NFT are decoupled. 
```

File: EIPS/eip-5791.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+- Verification of authenticity of the physical item requires action from a trusted entity (e.g. StockX).
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:24]:
```suggestion
- Verifying the authenticity of the physical item requires action from a trusted 3rd party (e.g. StockX).
```

File: EIPS/eip-5791.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+This document proposes a solution to mitigate these issues in a decentralized way. Its intent is to be the simplest possible path towards linking physical items to digital NFTs without a centralized authority.
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:26]:
```suggestion
```
Should be in the Rationale

File: EIPS/eip-5791.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Requirements
+
+This approach assumes that the physical item must have a chip attached to it that fulfills the following requirements:
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:34]:
```suggestion
This approach requres that the physical item must have a chip attached to it that fulfills the following requirements:
```

File: EIPS/eip-5791.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+- the ability to securely generate and store an ECDSA secp256k1 asymmetric keypair
+- the ability to sign messages from the private key of the asymmetric keypair
+- the ability for one to retrieve only the public key of the asymmetric keypair (private key non-extractable)
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:38]:
```suggestion
- The chip can securely generate and store an ECDSA secp256k1 asymmetric keypair;
- The chip can sign messages using the private key of the previously-generated asymmetric keypair;
- The chip exposes the public key; and
- The private key cannot be extracted
```

File: EIPS/eip-5791.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    event PBTMint(uint256 indexed tokenId, address indexed chipAddress);
+
+    /// @notice Emitted when a token is mapped to a different chip.
+    /// Chip replacements may be useful in certain scenarios (e.g. chip defect).
+    event PBTChipRemapping(uint256 indexed tokenId, address indexed oldChipAddress, address indexed newChipAddress);
+}
+
+```
+
+To aid recognition that an [EIP-721](./eip-721.md) token implements physical binding via this EIP: upon calling [EIP-721](./eip-721.md)’s `function supportsInterface(bytes4 interfaceID) external view returns (bool)` with `interfaceID=0xa70d2657`, a contract implementing this EIP must return true.
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:102]:
```suggestion
To aid recognition that an [EIP-721](./eip-721.md) token implements physical binding via this EIP: upon calling [EIP-165](./eip-165.md)’s `function supportsInterface(bytes4 interfaceID) external view returns (bool)` with `interfaceID=0xa70d2657`, a contract implementing this EIP must return true.
```

File: EIPS/eip-5791.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+### Out of Scope
+
+The following are some peripheral problems that are intentionally not within the scope of this EIP:
+
+- trusting that a specific NFT collection's chip addresses actually map to physical chips embedded in items, instead of arbitrary EOAs
+- ensuring that the chip does not deterioriate or get damaged
+- ensuring that the chip stays attached to the physical item
+- etc.
+
+Work is being done on these challenges in parallel. For example, the Ethereum Reality Service (`https://www.ers.to/`) is meant to tackle some of these issues.
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:119]:
```suggestion
Work is being done on these challenges in parallel.
```

File: EIPS/eip-5791.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+Additionally, it's possible for the owner of the physical item to transfer the NFT to a wallet owned by somebody else (by sending a chip signature to that other person for use). We still consider the NFT physical backed, as ownership management is tied to the physical item. This can be interpreted as the item's owner temporarily lending the item to somebody else, since (1) the item's owner must be involved for this to happen as the one signing with the chip, and (2) the item's owner can reclaim ownership of the NFT at any time.
+
+## Backwards Compatibility
+
+This proposal is backward compatible with [EIP-721](./eip-721.md) on an API level. As mentioned above, for the token to be physical-backed, the contract must use a account-bound implementation of [EIP-721](./eip-721.md) (all [EIP-721](./eip-721.md) functions that transfer must throw) so that transfers go through the new function introduced here, which requires a chip signature.
+
+## Reference Implementation
+
+At the time of writing, Kong is a supplier of chips that satisfy the cryptographic properties listed above.
+
```
Comment (Pandapip1) [inline EIPS/eip-5791.md:132]:
```suggestion
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Comment by 2pmflow** [comment]
@Pandapip1 thanks for detailed review. Feedback addressed in latest

File: EIPS/eip-5791.md (hunk 8 above)
Comment (2pmflow) [reply EIPS/eip-5791.md]: Was also wondering what the downsides were of including this link. It's a relatively light landing page that mostly redirects to https://ethereum-magicians.org/t/a-proposal-for-the-ethereum-reality-service/9694. Would a link to ethereum-magicians directly be more appropriate?

**Review by Pandapip1 — APPROVED** [review]

**Comment by kanedaaaa** [comment]
Can't wait to learn about the project and make some contributions to the repo. Very excited about this one.

**Pull request merged**